blocks/form/form.js

```diff
@@ -73,24 +125,30 @@ function renderField(fd) {
     field = createFieldWrapper(fd);
     field.append(createInput(fd));
   }
+  if (fd.Description) {
+    field.append(createHelpText(fd));
+  }
   return field;
 }
 
-async function fetchData(url) {
+async function fetchData(url, getId) {
   const resp = await fetch(url);
   const json = await resp.json();
-  return json;
+  return json.data.map((fd) => ({
+    ...fd,
+    Id: fd.Id || getId(fd.Name),
+  }));
 }
 
-async function fetchForm(pathname) {
+async function fetchForm(pathname, getId) {
```

Comment:
passing `idGenerator` through is strange, when you can just use the function as a singleton and put the map of ids into a closure.

Reply:
I agree. Not sure what i was thinking

Reply:
resolved.
https://github.com/vdua/nedbank/pull/1/files#diff-14d0faad899dcfcb4e5cd318658f4d58b9e2aa66ebc5912bbe09302339f6a564R162